src/java.base/share/classes/java/util/zip/ZipOutputStream.java

```diff
@@ -399,6 +403,10 @@ public void finish() throws IOException {
 
     /**
      * Closes the ZIP output stream as well as the stream being filtered.
+     * <p>
+     * A ZipException will be thrown if the combined length of the entry name,
+     * the extra field data, the entry comment and {@linkplain #CENHDR CEN Header size},
+     * exceeds 65,535 bytes.
```

Comment:
Is this missing text to say that close may write as part of closing?

Reply:
ZipOutputStream::close() calls DeflaterOutputStream::close() will in turn will call ZipOutputStream::finish()

I could remove the above and just leave the verbiage in ZipOutputStream::finish(), I only added it to close as I didn't think it was obvious that close resulted in a call to finish.

I do not have a strong preference or do you have an alterantive suggestion?

Reply:
I think having ZOS.close say that it finishes writing the contents of the ZIP output stream and closes it, would make the API docs easier to read. If you do then no need for close to talk about the ZipException.

Reply:
Is this what you were thinking:

 
```
     /**
-     * Closes the ZIP output stream as well as the stream being filtered.
+     * Finishes writing the contents of the ZIP output stream. The
+     * underlying stream will also be closed as well as the stream being filtered.
      * <p>
-     * A ZipException will be thrown if the combined length of the entry name,
-     * the extra field data, the entry comment and {@linkplain #CENHDR CEN Header size},
-     * exceeds 65,535 bytes.
      * @throws    ZipException if a ZIP file error has occurred
      * @throws    IOException if an I/O error has occurred
      */
```
And leaving the verbiage in finish() as is?